src/telalogin/token.ts

```diff
@@ -1,23 +1,32 @@
 import jwt = require('jsonwebtoken');
-import { QueryResult } from 'pg';
-import { stringify } from 'querystring';
 //import jwt from 'jsonwebtoken';
 
 const secret = 'umafrasequalquerparateste';
 export function setToken(email: string): string {
   const payload = {
     "email": email
   }
-  return jwt.sign(payload, secret, {expiresIn: 5});
+  return jwt.sign(payload, secret, {expiresIn: 300}); //5 min
 }
 
-export function verifyToken(token: string): string {
-  const obj = jwt.verify(token, secret, function(error, decoded) {
-    if (error)
+export function verifyToken(req, res): boolean {
+
+  const token: string = req.headers.authorization;
+  const splitToken = (token.split(" ", 2))[1];
+
+  let answer: boolean = false;
+
+  jwt.verify(splitToken, secret, function(error, decoded) {
+    if(error) {
+      res.status(401);
+      answer = false;
       return error;
+    }
     else {
+      answer = true;
       return decoded;
     }
-  })
-  return JSON.stringify(obj);
+  });
+  return answer;
```

Comment:
Acredito que essa lógica não funciona porque da forma que a função `verify` funciona se um callback é passado como terceiro parâmetro a função funciona como assincrona (https://github.com/auth0/node-jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback), então o answer sempre vai ser false porque o código ainda não passou pelo callback. 

Na minha opinião a melhor forma de fazer essa lógica é retornar uma Promise com o resultado ou passar um callback para a função e o callback ser chamado quando o token for decodificado.